jupyter_server/serverapp.py

```diff
@@ -187,21 +188,21 @@ class ServerWebApplication(web.Application):
 
     def __init__(self, jupyter_app, default_services, kernel_manager, contents_manager,
                  session_manager, kernel_spec_manager,
-                 config_manager, extra_services, log,
+                 config_manager, authorization_manager, extra_services, log,
```

Comment:
Do we consider these function signatures a part of the API? Does this trigger a major version bump?

Reply:
Good question.  I'm not finding any way that a subclass of `ServerWebApplication` could be used, but I wonder if we should minimize the issue by moving `authorization_manager` to the end and make it optional?  If so, I think we'd need to do similar with `iniit_settings()` below.  It kinda looks like that might have been the case with `jinja_env_options` actually.

Reply:
> authorization_manager to the end and make it optional?

yes, moving at the end, but the default would be a authorization_manager that would always return True. The security teams would say "always return False" by default...

Reply:
I think the backward-compatible approach is:

1. allow it to be omitted
2. if omitted, use NOPAuthorizationManager (preserves current behavior for backward compatibility)
3. warn about it, since a permissive choice is being made without user input